src/Psalm/Internal/Provider/ParserCacheProvider.php

```diff
@@ -184,7 +190,29 @@ private function getExistingFileContentHashes(): array
             $file_hashes_path = $root_cache_directory . DIRECTORY_SEPARATOR . self::FILE_HASHES;
 
             if ($root_cache_directory && is_readable($file_hashes_path)) {
+                $fp = fopen($file_hashes_path, 'r');
+                $max_wait_cycles = 5;
+                $has_lock = false;
+                while ($max_wait_cycles > 0) {
+                    if (flock($fp, LOCK_SH)) {
+                        $has_lock = true;
+                        break;
+                    }
+                    $max_wait_cycles--;
+                    usleep(50000);
+                }
+
+                if (!$has_lock) {
+                    fclose($fp);
+                    error_log('Could not acquire lock for content hashes file');
+                    $this->existing_file_content_hashes = [];
+
+                    return [];
+                }
+
+                // file_get_contents ignores any locks and gets the file, but we have the lock on $fp anyway
```

Comment:
Why not use fgets then? file_get_contents would have to reopen a stream, no?

Reply:
Changed to fread now